skimage/draw/draw.py

```diff
@@ -272,3 +268,354 @@ def set_color(img, coords, color, alpha=1):
+           [  0, 211,  43,   0,   0,   0,  43, 211,   0,   0],
+           [  0,  60, 194,  43,   0,  43, 194,  60,   0,   0],
+           [  0,   0,  60, 211, 255, 211,  60,   0,   0,   0],
+           [  0,   0,   0,   0,   0,   0,   0,   0,   0,   0],
+           [  0,   0,   0,   0,   0,   0,   0,   0,   0,   0]], dtype=uint8)
+    """
+    return _circle_perimeter_aa(cy, cx, radius, shape)
+
+
+def ellipse_perimeter(cy, cx, yradius, xradius, orientation=0, shape=None):
+    """Generate ellipse perimeter coordinates.
+
+    Parameters
+    ----------
+    cy, cx : int
+        Centre coordinate of ellipse.
+    yradius, xradius : int
+        Minor and major semi-axes. ``(x/xradius)**2 + (y/yradius)**2 = 1``.
+    orientation : double, optional
+        Major axis orientation in clockwise direction as radians.
+    shape : tuple, optional
+        Image shape which is used to determine the maximum extent of output pixel
+        coordinates. This is useful for ellipses which exceed the image size.
+        If None, the full extent of the ellipse are used.
+
+    Returns
+    -------
+    rr, cc : (N,) ndarray of int
+        Indices of pixels that belong to the ellipse perimeter.
+        May be used to directly index into an array, e.g.
+        ``img[rr, cc] = 1``.
+
+    References
+    ----------
+    .. [1] A Rasterizing Algorithm for Drawing Curves, A. Zingl, 2012
+           http://members.chello.at/easyfilter/Bresenham.pdf
+
+    Examples
+    --------
+    >>> from skimage.draw import ellipse_perimeter
+    >>> img = np.zeros((10, 10), dtype=np.uint8)
+    >>> rr, cc = ellipse_perimeter(5, 5, 3, 4)
+    >>> img[rr, cc] = 1
+    >>> img
+    array([[0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+           [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+           [0, 0, 0, 1, 1, 1, 1, 1, 0, 0],
+           [0, 0, 1, 0, 0, 0, 0, 0, 1, 0],
+           [0, 1, 0, 0, 0, 0, 0, 0, 0, 1],
+           [0, 1, 0, 0, 0, 0, 0, 0, 0, 1],
+           [0, 1, 0, 0, 0, 0, 0, 0, 0, 1],
+           [0, 0, 1, 0, 0, 0, 0, 0, 1, 0],
+           [0, 0, 0, 1, 1, 1, 1, 1, 0, 0],
+           [0, 0, 0, 0, 0, 0, 0, 0, 0, 0]], dtype=uint8)
+
+    """
+    return _ellipse_perimeter(cy, cx, yradius, xradius, orientation, shape)
+
+
+def bezier_curve(y0, x0, y1, x1, y2, x2, weight, shape=None):
+    """Generate Bezier curve coordinates.
+
+    Parameters
+    ----------
+    y0, x0 : int
+        Coordinates of the first control point.
+    y1, x1 : int
+        Coordinates of the middle control point.
+    y2, x2 : int
+        Coordinates of the last control point.
+    weight : double
+        Middle control point weight, it describes the line tension.
+    shape : tuple, optional
+        Image shape which is used to determine the maximum extent of output
+        pixel coordinates. This is useful for curves which exceed the image
+        size. If None, the full extent of the curve are used.
+
+    Returns
+    -------
+    rr, cc : (N,) ndarray of int
```

Comment:
If that's correct, it means that the API is wrong (see example below...). I vote to fix that elsewhere.